### ISSUE TITLE
useCookie doesn't work after an await despite using asyncContext experimental feature

### ISSUE THREAD

**Issue by joffreyBerrier** [nuxt/nuxt#23777]
### Environment

- Operating System: Darwin
- Node Version:     v19.9.0
- Nuxt Version:     3.7.4
- CLI Version:      3.9.0
- Nitro Version:    2.6.3
- Package Manager:  npm@9.8.0
- Builder:          -
- User Config: devtools, experimental
- Runtime Modules:  -
- Build Modules:    -

I'm using `"nuxt": "npm:nuxt3@latest"` 

The right version of nuxt is : `3.8.0-28284309.b3d3d7f4`

### Reproduction

https://github.com/joffreyBerrier/nuxt-use-cookie

### Describe the bug

Using **useCookie** after an await doens't work despite the fact that feature is available `experimental { asyncContext: true }`

**Error message**

```
error Error: [nuxt] A composable that requires access to the Nuxt instance was called outside of a plugin, Nuxt hook, Nuxt middleware, or Vue setup function. This is probably not a Nuxt bug. Find out more at `https://nuxt.com/docs/guide/concepts/auto-imports#using-vue-and-nuxt-composables`.
    at createError (index.mjs?v=164b9a79:78:15)
    at createError (error.js?v=164b9a79:31:16)
    at NuxtError (revive-payload.client.js?v=164b9a79:9:24)
    at hydrate (parse.js?v=164b9a79:60:32)
    at hydrate (parse.js?v=164b9a79:125:19)
    at hydrate (parse.js?v=164b9a79:125:19)
    at hydrate (parse.js?v=164b9a79:60:40)
    at unflatten (parse.js?v=164b9a79:132:9)
    at parse (parse.js?v=164b9a79:16:9)
    at parsePayload (payload.js?v=164b9a79:106:10)
```

**Comment by danielroe** [comment]
You should cause side effects within async data handlers (as it's not guaranteed to run, for example if using cached data).

We could consider making the nuxt context available within handlers, however.

**Comment by joffreyBerrier** [comment]
Hi @danielroe problem solved in **Nuxt 3.8.0** magic release  🪄
